checker/votour.ml

```diff
@@ -20,14 +19,14 @@ let rec read_num max =
   if l = "u" then None
   else if l = "x" then quit ()
   else
-    try
-      let v = int_of_string l in
+    match int_of_string l with
+    | v ->
       if v < 0 || v >= max then
         let () =
           Printf.printf "Out-of-range input! (only %d children)\n%!" max in
         read_num max
       else Some v
-    with Failure "int_of_string" ->
+    | exception Failure _ ->
```

Comment:
I think the code is fine, howeve I am not sure what OCaml upstream is thinking here, imagine we had two legacy functions that can raise different `Failure` values, how are we supposed to catch them ? The whole warning looks really fishy to me.

If they wanted to change the value, they should rather provide a new version of `Pervasives` raising the new exception instead of this bogus warning.